### PR TITLE
Poly2TriTriangulator fixes for DistributedMesh

### PR DESCRIPTION
**Pull request by roystgnr**
This is the set of changes needed to fix our Poly2Tri unit tests (and the tests in https://github.com/idaholab/moose/pull/21303) with `--enable-distmesh`, but that `MeshSerializer` update ought to make the serializer potentially more flexible for other use cases too.

### CODE REVIEW

**Comment by roystgnr** [comment]
Well, I spoke too soon.  Half the proposed MOOSE poly2tri tests now work with DistributedMesh, but I'll need to look deeper into the rest.  This ought to be merged regardless so we can get `--enable-distmesh` tests in our regular CI here.

**Comment by roystgnr** [comment]
Okay, with these last few commits (plus changes in MOOSE) my tests there are also working on DistributedMesh, and our unit test suite here is find with `--enable-distmesh` builds up to 8 ranks.  I'm still seeing something weird from an Exodus test at 9 ranks, but that looks like a separate issue so I'd like to merge this once CI is happy.

File: src/mesh/poly2tri_triangulator.C [hunk 1]
```diff
@@ -164,6 +164,11 @@ Poly2TriTriangulator::~Poly2TriTriangulator() = default;
 // Primary function responsible for performing the triangulation
 void Poly2TriTriangulator::triangulate()
 {
+  // We only operate on serialized meshes.  And it's not safe to
+  // serialize earlier, because it would then be possible for the user
+  // to re-parallelize the mesh in between there and here.
+  MeshSerializer _serializer(_mesh);
```
Comment (jwpeterson) [inline src/mesh/poly2tri_triangulator.C:170]: Leading underscore naming should be reserved for member variables.

**Review by jwpeterson — APPROVED** [review]

File: src/mesh/poly2tri_triangulator.C (hunk 1 above)
Comment (roystgnr) [reply src/mesh/poly2tri_triangulator.C:170]: Yup.  And this originally *was* a member variable, so I probably just cut and pasted.  Will fix.

**Comment by roystgnr** [comment]
If this passes I'll add some DistributedMesh sweeps to CI, and if *that* works too I'll merge.

**Comment by roystgnr** [comment]
Well, at least it's a new example failing this time.

**Comment by roystgnr** [comment]
Looks like my fixes for Triangle+PSLG+DistributedMesh failed horribly with Triangle+GENERATE_CONVEX_HULL+anything.

**Pull request merged**